py3status/modules/usbguard.py

```diff
@@ -1,195 +1,77 @@
+    {'full_text': '[Allow]', 'index': '22/allow', 'urgent': True}
 ]
 """
 
-from copy import deepcopy
 from threading import Thread
-from gi.repository import Gio
+from gi.repository import GLib, Gio
 import re
 
 STRING_USBGUARD_DBUS = "start usbguard-dbus.service"
-STRING_INVALID_FILTER = "invalid filters `{}`"
 
 
 class Py3status:
     """
     """
 
     # available configuration parameters
-    allow_urgent = False
-    button_next = 4
-    button_previous = 5
-    filters = ["None", "any", "allow", "block"]
-    format = (
-        "[{format_device_list} ]{format_button_device} {format_button_filter}"
-        "[\?color=darkgray&show \|]{format_button_permanent}"
-    )
-    format_button_allow = "[\?if=!policy=allow&color=good Allow]"
-    format_button_block = "[\?if=!policy=block&color=degraded Block]"
-    format_button_device = "[\?color=deepskyblue&show Device] {device}/{devices}"
-    format_button_filter = (
-        "[\?color=deepskyblue&show Filter ]"
-        "[\?if=filter=allow Allow|[\?if=filter=block "
-        "Block|[\?if=filter=None None|Any]]]"
-    )
-    format_button_permanent = (
-        "[\?if=permanent&color=white Permanently" "|\?color=darkgray Permanently]"
-    )
-    format_button_reject = "\?color=bad Reject"
-    format_device = (
-        "[{name}|Unknown {id} [\?color=darkgray {usb_id}]]"
-        "[ {format_button_allow}][ {format_button_block}]"
-    )
+    format = "[{format_device}]"
+    format_button_allow = "\[Allow\]"
+    format_button_reject = "\[Reject\]"
+    format_device = "{format_button_reject} [{name}|{usb_id}] {format_button_allow}"
     format_device_separator = " "
-    format_notification_message = "USB ID: {usb_id}\nName: {name}\nPort: {port}"
-    format_notification_title = (
-        "USB Device [\?if=policy=allow Allowed]"
-        "[\?if=policy=block Blocked]"
-        "[\?if=policy=reject Rejected]"
-    )
-    permanent = None
-
-    # playground
 
     def post_config_hook(self):
-        for _filter in self.filters:
-            if _filter not in ["None", "any", "allow", "block"]:
-                raise Exception(STRING_INVALID_FILTER.format(_filter))
-
         self.init = {
             "format_button": self.py3.get_placeholders_list(
                 self.format_device, "format_button_*"
             ),
-            "notifications": (
-                self.format_notification_title or self.format_notification_message
-            ),
-            "permanent": self.permanent is None,
-            "target": {"allow": 0, "block": 1, "reject": 2},
+            "target": {"allow": 0, "reject": 2},
         }
 
-        self.filter_index = 0
-        self.filter_length = len(self.filters)
-        self.device_index = 1
-        self.device_length = 0
-        self.permanent = self.permanent or False
-        self.cache_devices = {}
-        self.device_keys = [
+        self.keys = [
```

Comment:
I remember this part. See if we can `re.compile(a single long string)` instead of a list of keys to `re.compile`.